src/emoji/data.js

```diff
@@ -74,32 +73,31 @@ export const getFilteredEmojis = (
     const matchesEmojiName = Math.min(1, emoji.emoji_name.indexOf(query));
     const priority = matchesEmojiLiteral ? 0 : matchesEmojiName;
     if (priority !== -1) {
-      allMatchingEmoji.set(emoji.emoji_name, priority);
+      allMatchingEmoji.set(emoji.emoji_name, emoji);
     }
   }
 
   for (const emoji_name of Object.keys(activeImageEmojiByName)) {
     const priority = Math.min(1, emoji_name.indexOf(query));
     if (priority !== -1) {
-      allMatchingEmoji.set(emoji_name, priority);
+      allMatchingEmoji.set(emoji_name, {
+        emoji_type: 'image',
+        emoji_name,
+        emoji_code: activeImageEmojiByName[emoji_name].name,
+      });
```

Comment:
As I said in https://github.com/zulip/zulip-mobile/pull/5326#issuecomment-1089435130 , I'd like to avoid having an intermediate stage where we create these new emoji objects at query time. The way to fix that is to make this change (using the shared sort) after introducing the memoized selector.

Reply:
Please re-read my comment here: https://github.com/zulip/zulip-mobile/pull/5326#issuecomment-1090850356
The memoized selector can't be introduced while the priority sort logic is still in place, unless you want to have two selectors. And since this comment refers to an inefficiency in the code, that doesn't seem ideal (even if it is an intermediate commit).

Reply:
Specifically, if you introduce the new selector, now you have emoji objects. But you still need the priority value to do the old sort. So how to associate the priority with the object? Your suggestion was to use the object as the key, but that won't work as the map contains both unicode and image emojis, and the image emojis are meant to overwrite the unicode ones, except they can't when using the full object as a key. 
So you could still keep using the name as the key, but then you'd have to construct the final object again at a later point.

Reply:
> Your suggestion was to use the object as the key,

I see, I guess my previous comment wasn't clear. What I had in mind was to continue using the name as the key for the Map, but to have the values be like `[emoji, priority]` rather than just the priority.

Reply:
I'm going to push against this one a last time. Optimizing intermediate commits, in which the code in all likelyhood will never be executed, feels bad. I think in this case, it doesn't even help readability: your suggestion is to push the object into an array, inside another array. Not only does this seem convoluted (and expensive as well) but the resulting data structure does not even resemble anything in the previous or final state of the code.